src/System.Collections/tests/Generic/Dictionary/Dictionary.Generic.Tests.netcoreapp.cs

```diff
@@ -220,5 +222,197 @@ public void EnsureCapacity_Generic_CapacityIsSetToPrimeNumberLargerOrEqualToRequ
         }
 
         #endregion
+
+        #region TrimExcess
+
+        [Fact]
+        public void TrimExcess_Generic_NegativeCapacity_Throw()
+        {
+            var dictionary = new Dictionary<TKey, TValue>();
+            AssertExtensions.Throws<ArgumentOutOfRangeException>("capacity", () => dictionary.TrimExcess(-1));
+        }
+
+        [Fact]
+        public void TrimExcess_Generic_CapacitySmallerThanCount_Throws()
+        {
+            var dictionary = new Dictionary<TKey, TValue>();
+            dictionary.Add(GetNewKey(dictionary), CreateTValue(0));
+            AssertExtensions.Throws<ArgumentOutOfRangeException>("capacity", () => dictionary.TrimExcess(0));
+
+            dictionary = new Dictionary<TKey, TValue>(20);
+            dictionary.Add(GetNewKey(dictionary), CreateTValue(0));
+            AssertExtensions.Throws<ArgumentOutOfRangeException>("capacity", () => dictionary.TrimExcess(0));
+        }
+
+        [Fact]
+        public void TrimExcess_Generic_LargeInitialCapacity_TrimReducesSize()
+        {
+            var dictionary = new Dictionary<TKey, TValue>(20);
+            dictionary.TrimExcess(7);
+            Assert.Equal(7, dictionary.EnsureCapacity(0));
+        }
+
+        [Fact]
+        public void TrimExcess_Generic_TrimToLargerThanExistingCapacity_DoesNothing()
+        {
+            var dictionary = new Dictionary<TKey, TValue>();
+            int capacity = dictionary.EnsureCapacity(0);
+            dictionary.TrimExcess(20);
+            Assert.Equal(capacity, dictionary.EnsureCapacity(0));
+
+            dictionary = new Dictionary<TKey, TValue>(10);
+            capacity = dictionary.EnsureCapacity(0);
+            dictionary.TrimExcess(20);
+            Assert.Equal(capacity, dictionary.EnsureCapacity(0));
+        }
+
+        [Fact]
+        public void TrimExcess_Generic_DictionaryNotInitialized_CapacityRemainsAsMinPossible()
+        {
+            var dictionary = new Dictionary<TKey, TValue>();
+            Assert.Equal(0, dictionary.EnsureCapacity(0));
+            dictionary.TrimExcess();
+            Assert.Equal(0, dictionary.EnsureCapacity(0));
+        }
+
+        [Theory]
+        [InlineData(1)]
+        [InlineData(85)]
+        [InlineData(89)]
+        public void TrimExcess_Generic_ClearThenTrimNonEmptyDictionary_SetsCapacityTo3(int count)
+        {
+            Dictionary<TKey, TValue> dictionary = (Dictionary<TKey, TValue>)GenericIDictionaryFactory(count);
+            Assert.Equal(count, dictionary.Count);
+            // The smallest possible capacity size after clearing a dictionary is 3
+            dictionary.Clear();
+            dictionary.TrimExcess();
+            Assert.Equal(3, dictionary.EnsureCapacity(0));
```

Comment:
Why will it always be 3? Will it be a constant value over time?

Reply:
If dictionary is not yet initialized the capacity is zero, as soon as it is initialized it will get set to a prime number capacity.
In here, I am showing the smallest possible capacity size after Clear() and Trim() is 3, the smallest prime number.

The reason we don't set capacity to 1 I believe is because we are skipping some smaller prime numbers to be efficient (1, 2, 5, 13, etc. are skipped).
Because I think, if we were to set capacity to 1, then as soon as we add one item we need to increase capacity again which is costly.